First version

oraluben opened the pull request:
# Matrix

Matrix is a config fuzzer, i.e. generate a set of configurations for a program that need to be fuzzed.

oraluben committed:
First version

zhouqiang-cl commented on Makefile:
```diff
@@ -0,0 +1,34 @@
+GOBIN=$(shell go env GOBIN)
+endif
+
+PACKAGE_LIST := go list ./... | grep -vE "matrix/test|pkg/ptrace|zz_generated|vendor"
```
why pkg/ptrace

zhouqiang-cl commented on pkg/context/context.go:
```diff
@@ -0,0 +1,64 @@
+package context
+
```
Add licence?

zhouqiang-cl commented on pkg/context/context.go:
```diff
@@ -0,0 +1,64 @@
+	"chaos-mesh/matrix/pkg/node/data"
+	"chaos-mesh/matrix/pkg/random"
+	"chaos-mesh/matrix/pkg/serializer"
+	"fmt"
```
group it?

zhouqiang-cl commented on pkg/context/context.go:
```diff
@@ -0,0 +1,64 @@
+	return result
+}
+
+func simpleRecGen(hollow interface{}) interface{} {
```
what means hellow

you06 replied:
I think it's "trick or treat" which generates random variables.

zhouqiang-cl commented on pkg/context/scheduler.go:
```diff
@@ -0,0 +1 @@
+package context
```
Did we need this file?

zhouqiang-cl commented on pkg/node/abstract.go:
```diff
@@ -0,0 +1,11 @@
+	"chaos-mesh/matrix/pkg/serializer"
+)
+
+type AbstractConfig struct {
```
add comment?

zhouqiang-cl commented on pkg/node/abstract.go:
```diff
@@ -0,0 +1,11 @@
+type AbstractConfig struct {
+	Tag string
+	serializer.Config
+	Hollow interface{}
```
Hollow?

zhouqiang-cl commented:
please add a description

oraluben committed:
fmt

oraluben committed:
Add tiup example

oraluben committed:
Add licenses

oraluben committed:
Trivial cleanups

oraluben committed:
Fix tiup example

mahjonp reviewed (approved):
LGTM

oraluben committed:
Support default and fix uncovered choice branch

oraluben committed:
Finish pd config

oraluben committed:
Finish pd config with following configs:
MemQuotaQuery
OOMUseTmpStorage
OOMAction
TempStorageQuota
Performance.MaxMemory
Performance.StatsLease
Performance.FeedbackProbability
Performance.BindInfoLease
PreparedPlanCache.Capacity
PreparedPlanCache.MemoryGuardRatio
TiKVClient.GrpcConnectionCount
TiKVClient.CommitTimeout
TiKVClient.MaxBatchSize
TiKVClient.MaxBatchWaitTime
TiKVClient.CoprCache
StmtSummary.MaxStmtCount
StmtSummary.MaxSQLLength
StmtSummary.RefreshInterval
StmtSummary.HistorySize
